File: PIAA/Lab 4/Fibonacijev heap/Fibonacijev heap/Node.cpp
#include "Node.h"

Node::Node() {
	data = 0;
	next = nullptr;
	prev = nullptr;
	parent = nullptr;
	lmc = nullptr;
}

Node::Node(int d) {
	data = d;
	next = prev = parent = lmc = nullptr;
}

Node::~Node() {}

File: PIAA/Lab 5/Primov Algoritam/Primov Algoritam/Node.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Primov_Algoritam
{
    public class Node
    {
        public int Data { get; set; }
        public List<Edge> Adjacent { get; }
        public FNode? FNode { get; set; }
        public Node? Parent { get; set; }

        public Node()
        { 
            Data = 0;
            Adjacent=new List<Edge>();
        }

        public Node(int data)
        {
            Data = data;
            Adjacent= new List<Edge>();
        }

        public void AddAdjacency(Edge edge)
        {
            Adjacent.Add(edge);
        }

        public void ClearAdjacency()
        {
            Adjacent.Clear();
        }
    }
}
